Cargo.toml

```diff
@@ -4,15 +4,90 @@ members = [
     "fvm",
     "sdk",
     "shared",
+    "ipld/*",
+    "ipld/amt/fuzz",
+    "ipld/hamt/fuzz",
+    "ipld/kamt/fuzz",
+    "testing/calibration/shared",
     "testing/conformance",
+    "testing/common_fuzz",
+    "testing/common_fuzz/fuzz",
     "testing/integration",
-    "testing/calibration/shared",
-    "ipld/*",
     "testing/test_actors",
     "testing/test_actors/actors/*",
     "tools/fvm-bench",
 ]
 
+[workspace.package]
+version = "4.1.1"
+license = "MIT OR Apache-2.0"
+edition = "2021"
+repository = "https://github.com/filecoin-project/ref-fvm"
+authors = ["Protocol Labs", "Filecoin Core Devs"]
+
+[workspace.dependencies]
+# common
+serde = { version = "1.0.164", default-features = false, features = ["derive"] }
+thiserror = "1.0.40"
+anyhow = "1.0.71"
+rand = "0.8.5"
+rand_chacha = "0.3.0"
+serde_json = "1.0.99"
+serde_tuple = "0.5.0"
+byteorder = "1.4.3"
+hex = "0.4.3"
+num-traits = { version = "0.2.14", default-features = false }
+num-derive = "0.4.0"
+lazy_static = "1.4.0"
+log = "0.4.19"
+futures = "0.3.28"
+
+# IPLD/Encoding
+cid = { version = "0.10.1", default-features = false }
+multihash = { version = "0.18.1", default-features = false }
```

Comment:
It looks like this bumps the dependencies we were trying to control in the 3.2.0 branch, but it also looks like they've moved this into a workspace so that we can pull in just the specific libraries that we'd need.

@bridiver @cypt4 could we take a look at if it's possible to archive this fork now and use upstream directly? I think it may make it possible to reduce the number of overall transitive dependencies we bring in.